### PR TITLE
Addition of api gateway tests for create, get/list, delete

### PR DESCRIPTION
**Pull request by nwspeete-ibm**

### CODE REVIEW

**Comment by nwspeete-ibm** [comment]
@rabbah @mdeuser WIP gateway tests will be added here to be pulled down and tested against what is in progress

**Comment by rabbah** [comment]
please avoid using `WIP` in title and instead use `wip` label.

**Comment by nwspeete-ibm** [comment]
https://openwhisk-jenkins.swg-devops.com:8443/job/Playground2/796/ pg

**Comment by nwspeete-ibm** [comment]
@johnsolm travis passed, pg is good, this can be merged. Thanks Morgan!

File: tests/dat/apigw/testswaggerdoc1 [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                "responses": {
+                    "default": {
+                        "description": "Default response"
+                    }
+                },
+                "x-ibm-op-ext": {
+                    "actionName": "utils/echo",
+                    "actionNamespace": "whisk.system",
+                    "backendMethod": "POST",
+                    "backendUrl": "https://192.168.99.100/api/v1/namespaces/whisk.system/actions/utils/echo",
```
Comment (rabbah) [inline tests/dat/apigw/testswaggerdoc1]: will this work across all deployents?
this looks like a mac/docker host ip.

File: tests/dat/apigw/testswaggerdoc1 [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    "paths": {
+        "/test1": {
+            "get": {
+                "responses": {
+                    "default": {
+                        "description": "Default response"
+                    }
+                },
+                "x-ibm-op-ext": {
```
Comment (rabbah) [inline tests/dat/apigw/testswaggerdoc1]: why is this called "x-ibm-op-ext"?

File: tests/dat/apigw/testswaggerdoc1 (hunk 1 above)
Comment (rabbah) [inline tests/dat/apigw/testswaggerdoc1]: will this work across all deployents?
this looks like a mac/docker host ip.

File: tests/dat/apigw/testswaggerdoc1 [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                                    },
+                                    "to": {
+                                        "location": "body",
+                                        "name": "*"
+                                    }
+                                },
+                                {
+                                    "action": "insert",
+                                    "from": {
+                                        "value": "Basic I34EJL53W3HGP64AB4E4DJ37FFGA424238GF6J"
```
Comment (rabbah) [inline tests/dat/apigw/testswaggerdoc1]: what is this value?

File: tests/src/whisk/core/cli/test/ApiGwTests.scala [hunk 4]
```diff
@@ -60,11 +60,6 @@ class ApiGwTests
         val testurlop = "get"
         val testapiname = testName+" API Name"
         val actionName = testName+"_action"
-
-        // List result will look like:
-        // ok: APIs
-        // Action                            Verb             API Name  URL
-        // /_/CLI_APIGWTEST1_action          get  CLI_APIGWTEST1 API Name  http://172.17.0.1:9001/api/ab9082cd-ea8e-465a-8a65-b491725cc4ef/CLI_APIGWTEST1_bp/path
         try {
```
Comment (rabbah) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: why is there a try with no catch of finally?

File: tests/src/whisk/core/cli/test/ApiGwTests.scala [hunk 5]
```diff
@@ -79,4 +74,106 @@ class ApiGwTests
...
+        val actionName = testName+"_action"
+        try {
+            var rr = wsk.api.create(basepath = Some(testbasepath), relpath = Some(testrelpath), operation = Some(testurlop), action = Some(actionName), apiname = Some(testapiname))
+            rr.stdout should include("ok: created API")
+            rr = wsk.api.get(basepathOrApiName = Some(testapiname))
+            rr.stdout should include(testbasepath)
+            rr.stdout should include(s"${actionName}")
+        }
+        finally {
+            val deleteresult = wsk.api.delete(basepathOrApiName = testbasepath,expectedExitCode = DONTCARE_EXIT)
```
Comment (rabbah) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: shouldn't you check that the delete result exit code is either not found or success?

Comment (nwspeete-ibm) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: Added.

File: tests/src/whisk/core/cli/test/ApiGwTests.scala (hunk 4 above)
Comment (nwspeete-ibm) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: Finally exists there.

File: tests/dat/apigw/testswaggerdoc1 (hunk 2 above)
Comment (nwspeete-ibm) [inline tests/dat/apigw/testswaggerdoc1]: @mdeuser

File: tests/src/whisk/core/cli/test/ApiGwTests.scala (hunk 4 above)
Comment (rabbah) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: :( fooled by the code review UI.

File: tests/src/whisk/core/cli/test/ApiGwTests.scala [hunk 6]
```diff
@@ -77,6 +72,114 @@ class ApiGwTests
         }
         finally {
             val deleteresult = wsk.api.delete(basepathOrApiName = testbasepath, expectedExitCode = DONTCARE_EXIT)
+            deleteresult.stdout should include("ok: deleted API")
```
Comment (rabbah) [inline tests/src/whisk/core/cli/test/ApiGwTests.scala]: If you expect the delete to always succeed, a better alternative is to remove `expectedExitCode = DONTCARE_EXIT`.

**Comment by nwspeete-ibm** [comment]
@rabbah after new pg and travis pass, any changes left to make on this?

**Comment by nwspeete-ibm** [comment]
Passed in pg https://openwhisk-jenkins.swg-devops.com:8443/job/Playground2/814/

**Pull request merged**